docs/provider_integrations.md

```diff
@@ -0,0 +1,266 @@
+* **User**: Provides authentication of patrons, updating credentials and potentially creation of new patrons. Patrons are integrated with the Drupal user system.
+* **User consent**: Provides management of user consent data. If certain functionality requires explicit consent from a patron this allows the consent to be stored in a business system.
+* **Wayf**: Provides integration with single sign-on services for authentication based on the architecture of WAYF, a service fully provided by Danish e-Infrastructure Cooperation, under the Danish government's Agency for Science, Technology and Innovation. 
+
+## Provider requirements
+
+While it is possible to mix and match provider systems and provider types as needed by the individual library organisation using Ding2 there are some general requirements and patterns regarding how systems are integrated.
+
+### General requirements
+
+* A provider system must expose an API for the required functionality and data. Examples of protocols used for current integrations:
+  * HTTP + JSON/XML
+  * SOAP
+* It must be possible to consume the API using PHP as that is the programming language used by Ding2.  
+* The API should use some form of access control which allows it to be accessed by the Ding2 website and prevents access from outsiders. Examples of currently used authentication methods:
+  * Authentication (preferred) e.g. based on username/password or tokens. 
+  * Restriction by IP.
+* The API should be well-documented in regards to functionality and data exposed. Examples of documentation used by current provider systems:
+  * OpenAPI
+  * SOAP 
+* The API should provide detailed error information which allows the website to provide sufficient feedback to the user.
+* The API should respond fast enough to be called synchronously by the website.
+
+### Library system requirements
+
+We refer to a system responsible for managing patrons, materials and material instances and the relationships between these as a library system. When used as a provider system it usually provides the following types:
+
+* User
+* Availability
+* Loan
+* Reservation
+* Debt
+* Historical loans (optional)
+* User consent (optional)
+
+#### User
+
+It must be possible to authenticate a patron given a username and a password. The username can also be in the form of a patron number, social security number or other forms of textual identification. The password can also be a PIN code.
+
+The authentication API should return some form of method for identifying the patron for subsequent requests. It is preferable not to store authentication credentials for each user within the Ding2 system after login procedure has completed.
+
+The authentication API may provide information about whether a patron is blocked i.e. prevented from accessing the library based on debts, bad behaviour or the like.
+
+Given a new password it must be possible to change the password of a patron using the API.
```

Comment:
Is this really a requirement? The unilogin integration for eReolen doesn't allow password change, and I don't think wayf does either.